### PR TITLE
Restore QueryId to two access control checks

### PR DESCRIPTION
**Pull request by ksobolew**
## Description

Commit 309f203804fb2bb12573d18d157b554c0b9fa3c5 made the `QueryId` mandatory, instead of `Optional`, in `SystemSecurityContext` and removed the security context from invocations which can't reliably pass it. It was also removed from these invocations, but not because it wasn't available, but because nobody actually passed it. This commit series brings it back into the SPI and makes the engine pass it to access control.

## Additional context and related issues

NOTE: It does not restore the full `SystemSecurityContext`, because it requires a `TransactionId`, which is truly not available. At least for `checkCanExecuteQuery` it is not, AFAICT.

## Release notes

( ) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
(x) Release notes are required, with the following suggested text: `checkCanExecuteQuery` and `checkCanSetSystemSessionProperty` checks of `SystemAccessControl` now receive `QueryId`.

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: core/trino-spi/src/main/java/io/trino/spi/security/SystemAccessControl.java [hunk 1]
```diff
@@ -204,12 +204,23 @@ default void checkCanWriteSystemInformation(Identity identity)
      * Check if identity is allowed to set the specified system property.
      *
      * @throws AccessDeniedException if not allowed
+     * @deprecated use {@link #checkCanSetSystemSessionProperty(Identity, QueryId, String)}
```
Comment (findepi) [inline core/trino-spi/src/main/java/io/trino/spi/security/SystemAccessControl.java:207]: make sure to follow-up with removal of the deprecated method

File: core/trino-spi/src/main/java/io/trino/spi/security/SystemAccessControl.java [hunk 2]
```diff
@@ -128,12 +129,23 @@ default void checkCanSetUser(Optional<Principal> principal, String userName)
      * Checks if identity can execute a query.
      *
      * @throws AccessDeniedException if not allowed
+     * @deprecated use {@link #checkCanExecuteQuery(Identity, QueryId)}
```
Comment (findepi) [inline core/trino-spi/src/main/java/io/trino/spi/security/SystemAccessControl.java:132]: make sure to follow-up with removal of the deprecated method

Comment (findepi) [inline core/trino-spi/src/main/java/io/trino/spi/security/SystemAccessControl.java:132]: add `@Deprecated` annotation. is allows IDEs to see deprecation even if source/javadoc is not downloaded, and is likely required by many tools to observe the deprecation (javac?)

**Review by findepi — APPROVED** [review]

File: core/trino-spi/src/main/java/io/trino/spi/security/SystemAccessControl.java (hunk 2 above)
Comment (ksobolew) [reply core/trino-spi/src/main/java/io/trino/spi/security/SystemAccessControl.java:132]: Forgot about it, thanks

File: core/trino-spi/src/main/java/io/trino/spi/security/SystemAccessControl.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (findepi) [reply core/trino-spi/src/main/java/io/trino/spi/security/SystemAccessControl.java:207]: -> https://github.com/trinodb/trino/pull/21411

**Comment by findepi** [comment]
CI 
- https://github.com/trinodb/trino/issues/21413

and some other failure. @ksobolew please investigate

**Comment by ksobolew** [comment]
The other CI was #20309

**Pull request merged**